### PR TITLE
#70 Build fn overrides; adds renaming and skipping

### PR DESCRIPTION
**Pull request by TedDriggs**
Here's the skip and rename functionality described in #70.

### CODE REVIEW

File: derive_builder/CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 ## Unreleased
 
 ### Added
-- customize setter names via `#[builder(setter(name="..."))]`
```
Comment (colin-kiegel) [inline derive_builder/CHANGELOG.md]: please don't remove this line - it's still true. ;-)

File: derive_builder/src/options/struct_mode.rs [hunk 2]
```diff
@@ -40,6 +44,82 @@ impl StructMode {
...
+            }
+        }
+    }
+    
+    /// e.g `private` in `#[builder(build_fn(private))]`
+    fn parse_build_fn_options_word(&mut self, ident: &syn::Ident) {
+        trace!("Setter Options - Parsing word `{}`", ident.as_ref());
+        match ident.as_ref() {
+            "skip" => {
+                self.build_fn_enabled = false;
```
Comment (colin-kiegel) [inline derive_builder/src/options/struct_mode.rs]: please use `self.build_fn_enabled(false)` instead. I know I've not done that consistently myself, though. ^^

File: derive_builder/src/options/struct_mode.rs [hunk 3]
```diff
@@ -40,6 +44,82 @@ impl StructMode {
...
+    
+    fn parse_build_fn_name(&mut self, lit: &syn::Lit) {
+        trace!("Parsing build function name `{:?}`", lit);
+        let value = parse_lit_as_string(lit).unwrap();
+        self.build_fn_name(value.clone())
+    }
+    
+    #[allow(dead_code,unused_variables)]
+    fn parse_build_fn_skip(&mut self, skip: &syn::Lit) {
+        self.build_fn_enabled = !parse_lit_as_bool(skip).unwrap();
```
Comment (colin-kiegel) [inline derive_builder/src/options/struct_mode.rs]: please use `self.build_fn_enabled(/*...*/)` instead

File: derive_builder_core/src/build_method.rs [hunk 4]
```diff
@@ -184,4 +184,27 @@ mod tests {
             }
         ));
     }
+    
+    #[test]
+    fn skip() {
+        let mut build_method = default_build_method!();
+        build_method.enabled = false;
+        
+        assert_eq!(quote!(), quote!(#build_method));
```
Comment (colin-kiegel) [inline derive_builder_core/src/build_method.rs]: Could you change the order here for consistency, please?
* All current uses of `assert_eq!` in this crate have the form `assert_eq!(computed_value, expected_value)`.

However, we can talk about switching the argument order _in general_, if you have a strong preference here (maybe as a separate refactoring issue/PR then).

File: derive_builder/CHANGELOG.md (hunk 1 above)
Comment (TedDriggs) [inline derive_builder/CHANGELOG.md]: Oops! Merge problems :(

File: derive_builder_core/src/build_method.rs (hunk 4 above)
Comment (TedDriggs) [inline derive_builder_core/src/build_method.rs]: I like the ordering as-is, don't know why I deviated from it in this case.

File: derive_builder/src/options/struct_mode.rs (hunk 3 above)
Comment (TedDriggs) [inline derive_builder/src/options/struct_mode.rs]: Fixed.

File: derive_builder/src/options/struct_mode.rs (hunk 2 above)
Comment (TedDriggs) [inline derive_builder/src/options/struct_mode.rs]: Fixed.

**Comment by colin-kiegel** [comment]
cool, thank you. :-)

**Pull request merged**